packages/matter.js/src/codec/MessageCodec.ts

```diff
@@ -122,8 +131,14 @@ export class MessageCodec {
 
         const sessionType = securityFlags & 0b00000011;
         if (sessionType !== SessionType.Group && sessionType !== SessionType.Unicast) throw new Error(`Unsupported session type ${sessionType}`);
-
-        return { sessionId, sourceNodeId, messageId, destGroupId, destNodeId, sessionType };
+        const hasPrivacyEnhancements = (securityFlags & SecurityFlag.HasPrivacyEnhancements) !== 0;
+        if (hasPrivacyEnhancements) throw new Error(`Privacy enhancements not supported`);
```

Comment:
Should this be MatterError rather than Error?

Reply:
The other errors Thrown here are also no MatterErrors. i will put on my list and do one big "all is MatterError" PR separately